### DISCUSSION TITLE
Tasmota script, smartmeter & resol deltasol M

### DISCUSSION THREAD

**Discussion by ppernias** [arendst/Tasmota#17341] [Q&A] [answered]
I've a Deltasol M controller and I would like to use the astonishing Tasmota Smartmeter firmware in order to decode the data stream (using the VBUS resol protocol) but my achievements are not fully ok.
I've decoded the temperature sensors connected to the deltasol M but I can't obtain the status of the pumps.
This is the decoding script I'm using (only decode the sensors/pumps connected)

>D
r="1,AA100011731000011248"
>B
=>sensor53 r
=>sensor53 d0
>M 1
+1,3,v,0,9600,Solar
%r%vo0sw@10,S1 ,°C,t_col,1
%r%vo2sw@10,S2 ,°C,t_dep1,1
%r%vo4sw@10,S3 ,°C,t_retorno_acs,1
%r%vo6sw@10,S4 ,°C,t_dep2a,1
%r%vo12sw@10,S7 ,°C,t_dep2b,1
%r%vo14sw@10,S8 ,°C,retorno_sr,1
%r%vo44ub@1,pump1 ,%%,pump1,0
%r%vo45ub@1,pump2 ,%%,pump2,0
%r%vo46ub@1,pump3 ,%%,pump3,0
%r%vo47ub@1,pump4 ,%%,pump4,0
%r%vo49ub@1,pump6 ,%%,pump6,0
#

and this is the result:
![Captura de pantalla 2022-12-10 a las 18 52 15](https://user-images.githubusercontent.com/6562572/206868655-aeacde95-0a23-43dd-aadc-a212bc36222e.png)

I've consulted previous discussions but all of them are about more reciente controllers.

By the way, the dump of the hex code (using the same as above but with "sensor53 d1") is:

18:57:14.323 Script: performs "sensor53 d1"
18:57:14.328 MQT: stat/d1_resol/RESULT = {"Time":"2022-12-10T18:57:14","SML":{"CMD":"dump: 1"}}
18:57:14.524 
18:57:14.857 : aa 10 00 11 73 10 00 01 12 48 7b 01 56 01 00 2c 60 00 02 01 05 17 38 22 38 22 05 46 41 01 37 01 00 05 38 22 38 22 05 46 
18:57:15.524 : aa 00 00 11 73 20 00 05 00 00 00 00 00 00 00 56 
18:57:15.858 : aa 10 00 11 73 10 00 01 12 48 7a 01 56 01 00 2d 5f 00 02 01 05 18 38 22 38 22 05 46 41 01 37 01 00 05 38 22 38 22 05 46 
18:57:16.524 : aa 00 00 11 73 20 00 05 00 00 00 00 00 00 00 56 

What I'm missing? The doc about vbus specifications helps me to decode the sensors but it is not ok for the rest of the data...
http://danielwippermann.github.io/resol-vbus/#/vsf/bytes/00_0010_7311_10_0100

**Comment by gemu2015** [comment, marked as answer]
#define SML_BSIZ 48
is the default for serial meter buffer size.
this is far too low for large data offsets like vo44
try to increase this value to e.g. 256

**Reply by ppernias** [reply]
Thanks! Now, it's decoding everything ok.
But I'm receiving some inconsistent data. (Some data are changing from 0 to 100 for a second randomly) 
I've tried:
#define TMSBSIZ 512
and just to be sure that I'm not reaching limits, 
#define MAX_METERS 1 
(I only have one meter)
Why the way,
In a few days I'll receive a PCB I've built in order to support the electronic in a proper way (better than in a protoboard) and I'll upload some pictures of the system running. Thanks!

**Comment by ppernias** [comment]
I've received the PCB. Just some pictures and the gerber files in the case somebody want to order. I ordered mine to JLPCB(very fast and cheap)
[Gerber_PCB_wemos_tasmota_resol.zip](https://github.com/arendst/Tasmota/files/10213039/Gerber_PCB_wemos_tasmota_resol.zip)
![Photo View_2022-12-13](https://user-images.githubusercontent.com/6562572/207189703-5c1de2cb-7f6d-49bb-bcb5-28b2c52581a7.svg)
![Photo View_2022-12-13 (1)](https://user-images.githubusercontent.com/6562572/207189896-2945ea94-2908-491b-95dc-d6b84c6f2a64.svg)

![Captura de pantalla 2022-12-13 a las 1 15 54](https://user-images.githubusercontent.com/6562572/207193428-5b12cca7-bec6-4af4-b437-cc2861130bbc.png)

I partially solved the problem of inconsistent data by changing the directive to use median instead direct values.
So, the final decode script is:
>D
r="1,AA100011731000011248"
>B
=>sensor53 r
=>sensor53 d0
>M 1
+1,3,v,16,9600,Solar
%r%vo0sw@10,S1 ,°C,t_col,1
%r%vo2sw@10,S2 ,°C,t_dep1,1
%r%vo4sw@10,S3 ,°C,t_retorno_acs,1
%r%vo6sw@10,S4 ,°C,t_dep2a,1
%r%vo12sw@10,S7 ,°C,t_dep2b,1
%r%vo14sw@10,S8 ,°C,retorno_sr,1
%r%vo44ub@1,pump1 ,%%,impulsion_tubos,0
%r%vo45ub@1,pump2 ,%%,v_dep1,0
%r%vo46ub@1,pump3 ,%%,v_caldera,0
%r%vo47ub@1,pump4 ,%%,v_dep_2,0
%r%vo49ub@1,pump6 ,%%,v_intercooler,0
%r%vo50ub@1,pump7 ,%%,v_retorno_acs,0
%r%vo66ut@#,time,,tiempo,1
#

NOTE: if you are using the gerber. Notice that sometimes the SOT92 print in the board does not correspond with the orientation of the capsule. Use E and C to place correctly your BC547B.

**Reply by bsp9493** [reply]
Nice Board! thanx for the files!

Unrelated: anyone know how do I get the device to display temps in F instead of C?
I have tried SetOption8 1 in the console, but temps still appear in C

Thanx in advance
